src/components/ForgotPassword.tsx

```diff
@@ -0,0 +1,356 @@
+  const [codeError, setCodeError] = useState(false);
+  const [codeBackendError, setCodeBackendError] = useState(false);
+  const inputs = useRef<(HTMLInputElement | null)[]>([]);
+  const [error, setError] = useState("");
+  const [passwordError, setPasswordError] = useState(false);
+  const [counter, setCounter] = React.useState(30);
+
+  const handleEmailSubmit = () => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    const emailFound = true; // Placeholder, replace with backend logic
+    if (!emailRegex.test(email.trim())) {
+      setError("Please enter a valid email address.");
+    } else if (!emailFound) {
+      setError("Couldn't find your account");
+    } else {
+      setError("");
+      setStep(2);
+      setCounter(30);
+    }
+  };
+
+  const handlePasswordSubmit = () => {
+    if (
+      confirmPassword !== newPassword ||
+      newPassword.trim() === "" ||
+      confirmPassword.trim() === ""
+    ) {
+      setPasswordError(true);
+    } else {
+      setPasswordError(false);
+      setStep(4);
+    }
+  };
+
+  const handleCodeSubmit = () => {
+    const isCodeCorrect = true; // Placeholder, replace with backend logic
+    if (code.some((digit) => digit === "")) {
+      setCodeError(true);
+      setCodeBackendError(false);
+    } else if (!isCodeCorrect) {
+      setCodeBackendError(true);
+      setCodeError(false);
+    } else {
+      setCodeError(false);
+      setCodeBackendError(false);
+      setStep(3);
+    }
+  };
+
+  const handleCodeChange = (value: string, index: number) => {
+    if (!/^\d$/.test(value) && value !== "") return;
+
+    const newCode = [...code];
+    newCode[index] = value;
+    setCode(newCode);
+
+    if (codeError || codeBackendError) {
+      setCodeError(false);
+      setCodeBackendError(false);
+    }
+
+    if (value && index < 3 && inputs.current[index + 1]) {
+      inputs.current[index + 1]?.focus();
+    }
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent, index: number) => {
+    if (
+      e.key === "Backspace" &&
+      !code[index] &&
+      index > 0 &&
+      inputs.current[index - 1]
+    ) {
+      (inputs.current[index - 1] as HTMLInputElement)?.focus();
+    }
+  };
+
+  const handleKeyDownSpace = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.code === "Space") {
+      e.preventDefault();
+    }
+  };
+
+  React.useEffect(() => {
+    let timer: NodeJS.Timeout | undefined;
+    if (counter > 0) {
+      timer = setInterval(() => setCounter((prev) => prev - 1), 1000);
+    }
+    return () => {
+      if (timer) {
+        clearInterval(timer);
+      }
+    };
+  }, [counter]);
+
+  const CustomInputProps = (inputType: "newPassword" | "confirmPassword") => {
+    const toggleVisibility = () => {
+      if (inputType === "newPassword") {
+        setShowNewPassword((prev) => !prev);
+      } else if (inputType === "confirmPassword") {
+        setShowConfirmPassword((prev) => !prev);
+      }
+    };
+
+    return {
+      endAdornment: (
```

Comment:
If the "New Password" field is empty then the show password button is not clickable